apps/chat/src/components/Common/ApplicationDialog.tsx

```diff
@@ -49,52 +49,42 @@ interface Props {
   currentReference?: string;
 }
 
-const validateUrl = (url: string) => {
-  const pattern = new RegExp(
-    '^(https?:\\/\\/)?' +
-      '(?:(?:[a-z\\d][a-z\\d-]*[a-z\\d])\\.)+[a-z]{2,}|' +
-      '((\\d{1,3}\\.){3}\\d{1,3})' +
-      '(\\:\\d+)?(\\/?[-a-z\\d%_.~+]*)*' +
-      '(\\?[;&a-z\\d%_.~+=-]*)?',
-  );
-
-  return pattern.test(url);
-};
-const getItemLabel = (item: string) => item;
-
-export const ApplicationDialog = ({
+export const ApplicationDialog: React.FC<Props> = ({
   isOpen,
   onClose,
-  isEdit,
   selectedApplication,
+  isEdit,
   currentReference,
-}: Props) => {
+}) => {
+  const {
+    register,
+    handleSubmit,
+    setValue,
+    clearErrors,
+    trigger,
+    control,
+    formState: { errors, isValid },
+  } = useForm<FormData>({
+    mode: 'onChange',
+    reValidateMode: 'onChange',
+  });
   const dispatch = useAppDispatch();
-  const { t } = useTranslation(Translation.PromptBar);
-  const [name, setName] = useState('');
-  const [version, setVersion] = useState('');
-  const [versionHasFocus, setVersionHasFocus] = useState(false);
-  const [featuresDataHasFocus, setFeaturesDataHasFocus] = useState(false);
-  const [description, setDescription] = useState('');
   const [deleteLogo, setDeleteLogo] = useState(false);
   const [localLogoFile, setLocalLogoFile] = useState<string | undefined>();
-  const files = useAppSelector(FilesSelectors.selectFiles);
-  const featuresDataInputRef = useRef<HTMLTextAreaElement>(null);
-  const [features, setFeatures] = useState('');
-  const [featuresDataError, setFeaturesDataError] = useState('');
-  const [maxAttachments, setMaxAttachments] = useState(0);
-  const [completionUrl, setCompletionUrl] = useState('');
-  const [completionUrlError, setCompletionUrlError] = useState('');
-  const [submitted, setSubmitted] = useState(false);
-  const [nameError, setNameError] = useState<string | null>(null);
-  const [versionError, setVersionError] = useState<string | null>(null);
-  const [iconError] = useState<string | null>(null);
-  const nameInputRef = useRef<HTMLInputElement>(null);
-  const descriptionInputRef = useRef<HTMLTextAreaElement>(null);
-  const [filterParams, setFilterParams] = useState<string[]>([]);
+  const [inputAttachmentTypes, setInputAttachmentTypes] = useState<string[]>(
+    [],
+  );
+  const [featuresInput, setFeaturesInput] = useState('');
   const [isDeleteModalOpen, setIsDeleteModalOpen] = useState(false);
   const [isPublishing, setIsPublishing] = useState(false);
 
+  const loading = useAppSelector((state) => state.application.loading);
+  const files = useAppSelector(FilesSelectors.selectFiles);
+  const { t } = useTranslation(Translation.PromptBar);
+  const getItemLabel = (item: string) => item;
+
+  const inputClassName = classNames('input-form peer mx-0', 'input-invalid');
+
   const entity = selectedApplication && {
```

Comment:
```suggestion
  const entity = selectedApplication ? {...} : null;
 ```
 
 Lets refactor it please to improve readability